plugins/pulp_rpm/plugins/importers/yum/parse/treeinfo.py

```diff
@@ -82,21 +84,22 @@ def sync(sync_conduit, feed, working_dir, nectar_config, report, progress_callba
         _LOGGER.debug('downloading distribution files')
         downloader.download(file_to_download_request(f, feed, tmp_dir) for f in files)
         if len(listener.failed_reports) == 0:
-            unit = sync_conduit.init_unit(ids.TYPE_ID_DISTRO, model.unit_key, model.metadata,
-                                          model.relative_path)
+            model.set_content(tmp_dir)
+            model.save()
+            # The save sets the content path, which is needed to generate the download_reports
+            # Long term this should be done by a serializer
             model.process_download_reports(listener.succeeded_reports)
-            # remove pre-existing dir
-            shutil.rmtree(unit.storage_path, ignore_errors=True)
-            pulp_copytree(tmp_dir, unit.storage_path)
-            # mkdtemp is very paranoid, so we'll change to more sensible perms
-            os.chmod(unit.storage_path, 0o775)
-            sync_conduit.save_unit(unit)
+            model.save()
+
+            repo_controller.associate_single_unit(repo, model)
+
             # find any old distribution units and remove them. See BZ #1150714
             for existing_unit in existing_units:
-                if existing_unit != unit:
+                if existing_unit != model:
                     _LOGGER.info("Removing out-of-date distribution unit %s for repo %s" %
                                  (existing_unit.unit_key, sync_conduit.repo_id))
-                    sync_conduit.remove_unit(existing_unit)
+                    platform_models.RepositoryContentUnit.objects(
+                        repo_id=sync_conduit.repo_id, unit_id=existing_unit.id).delete()
```

Comment:
With the next push this will look like:

```
                    platform_models.RepositoryContentUnit.objects(
                        repo_id=sync_conduit.repo_id, 
                        unit_type_id=models.Distribution._content_type_id.default, 
                        unit_id=existing_unit.id).delete()
```